specification/servicefabric/resource-manager/Microsoft.ServiceFabric/2016-09-01/servicefabric.json

```diff
@@ -356,6 +395,96 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.ServiceFabric/locations/{location}/environments/{environment}/clusterVersions/{clusterVersion}": {
+      "get": {
+        "tags": [
+          "ClusterVersion"
+        ],
+        "operationId": "ClusterVersions_GetByEnvironment",
+        "description": "Get cluster code versions by environment",
+        "x-ms-examples": {
+          "Get cluster version": {
+            "$ref": "./examples/ClusterVersionsGetByEnvironment_example.json"
+          }
+        },
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/locationForClusterCodeVersions"
+          },
+          {
+            "$ref": "#/parameters/environment"
+          },
+          {
+            "$ref": "#/parameters/api-version"
+          },
+          {
+            "$ref": "#/parameters/subscriptionId"
+          },
+          {
+            "$ref": "#/parameters/clusterVersion"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK - Get cluster code versions successfully",
+            "schema": {
+              "$ref": "#/definitions/ClusterCodeVersionsListResult"
```

Comment:
let me tell why, 
our service will return a version for windows and a version for Linux, if applicable.  so we need to return a list.